modeltranslation/tests/translation.py

```diff
@@ -85,6 +85,22 @@ class ForeignKeyModelTranslationOptions(TranslationOptions):
     )
 
 
+@register(models.FirstSetNullForeignKeyModel)
+class FirstSetNullForeignKeyModelTranslationOptions(TranslationOptions):
+    fields = (
+        "test",
+    )
+
+
+@register(models.SecondSetNullForeignKeyModel)
+class SecondSetNullForeignKeyModel(TranslationOptions):
+    fields = (
+        "test",
+    )
```

Comment:
I see the problem now.

You should register for translation your base model, if you're using fields from base model.

Please try it. And if it fixes the issue, we can raise error "ImproperlyConfigured()" or something like that.

Reply:
Yep, that would make it work. If my base model is not under my control, it will not. There probably would be a solution, but the Django code is very complex :|. Let me dig a bit more into...

Reply:
Sorry. Misunderstood. It's this section: https://django-modeltranslation.readthedocs.io/en/latest/registration.html#translationoptions-fields-inheritance

So I would register the abstract base model with it's foreign key for translation. And maye the sub models, with additional translated fields, right?

Reply:
Just found I got a third model, inheriting from base class, without having the specific field translated. That would not be possible anymore...but would probably not be a problem, besides having unused fields.

Reply:
Main point: you register fields for the model they're defined on.

I'm not sure, if you also need to register child models (with empty fields), try both variants and see which works.

Reply:
I've done this many times without problems (registering only the final model, but not the abstract base class). But yeah, I can see why it doesn't work in my edge case, with foreign key.

Reply:
None works.I've not found a solution, and provided a working (failing) test. So I guess it is a bug?

Reply:
Yeah, probably a bug.